pinc/Project.inc

```diff
@@ -452,19 +452,12 @@ class Project
             $errors[] = _("This project is checked out and must specify a PPer/PPVer");
         }
 
-        if ($this->postednum) {
-            if (! preg_match('/^[1-9][0-9]*$/', $this->postednum)) {
-                $errors[] = sprintf(
-                    _("Posted Number \"%s\" is not of the correct format."),
-                    $this->postednum
-                );
-                // Occasionally, there will be a PG ebook that is still
-                // under U.S. copyright. This is indicated in their system
-                // by appending a 'C' to the etext number. The link to
-                // the etext, however, does not include the 'C', nor should
-                // the DP link. If this changes, update the pattern here.
-            }
-        }
+        // $this->postednum is either null or an integer by type definition.
+        // Occasionally, there will be a PG ebook that is still
+        // under U.S. copyright. This is indicated in their system
+        // by appending a 'C' to the etext number. The link to
+        // the etext, however, does not include the 'C', nor should
+        // the DP link.
```

Comment:
Would it worth hoisting this comment next to the declaration of `$postednum` above now that we don't do any validation? This seems like a worthwhile gotcha about the field.

Reply:
I tried that and it just felt out of place in the definition. It feels a little more intuitive here where we are doing validation.